src/md/ceefilegen/cceegen.cpp

```diff
@@ -47,10 +47,20 @@ HRESULT CCeeGen::CreateNewInstance(CCeeGen* & pGen) // static, public
     TESTANDRETURNMEMORY(pGen->m_peSectionMan);
 
     HRESULT hr = pGen->m_peSectionMan->Init();
-    TESTANDRETURNHR(hr);
+    if (FAILED(hr))
+    {
+        pGen->Cleanup();
+        delete pGen;
```

Comment:
Can you please use NewHolder<CCeeGen> as a local and use its Extract method to the output parameter in the success case? It is preferred to use holders in the runtime, especially when there are multiple return code paths.